be/src/exec/spill/spilled_stream.cpp

```diff
@@ -36,16 +36,263 @@
 #include "util/blocking_queue.hpp"
 
 namespace starrocks {
-// TODO: implements SpilledFileStream
+// some config
+const size_t chunk_buffer_max_size = 2;
+
+// read column one-by-one
+class SequentialFileStream final : public SpilledInputStream {
+public:
+    SequentialFileStream(const SpillFormater& formater, std::vector<std::shared_ptr<SpillFile>> spilled_files)
+            : _formater(formater), _spilled_files(std::move(spilled_files)) {}
+    ~SequentialFileStream() override = default;
+    StatusOr<ChunkUniquePtr> read(SpillFormatContext& context) override;
+    bool is_ready() override { return false; }
+    void close() override;
+
+private:
+    size_t _current_idx{};
+    std::unique_ptr<RawInputStreamWrapper> _readable;
+    const SpillFormater& _formater;
+    const std::vector<std::shared_ptr<SpillFile>> _spilled_files;
+    size_t output_chunks{};
+};
+
+StatusOr<ChunkUniquePtr> SequentialFileStream::read(SpillFormatContext& context) {
+    const size_t max_eos_retry_times = 2;
+    size_t eos_retry_times = 0;
+    while (eos_retry_times++ < max_eos_retry_times) {
```

Comment:
why do we need to retry here?

Reply:
if restore_from_fmt return a eos, this loop won't read any chunk. it's not as expected

Reply:
Same question, It's weird to me.
Why not just using `while(true)`? It seems to work.

Reply:
> if the eos of previous file is encountered, switch to next file in this round. we can use a flag to indicate this situtation. retry implicts fault-tolerance , quite confusing.

I'll fix it later